sycl/doc/extensions/proposed/sycl_ext_oneapi_memcpy2D.asciidoc

```diff
@@ -0,0 +1,131 @@
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension adds support for 2D memcpy routines that can copy a specified
+rectangular region in the presence of array padding.
+
+The handler class gains one new method:
+
+[cols="^60a,40"]
+|===
+| Member Function | Description
+
+a|
+[source,c++]
+----
+void memcpy2D(void *dest, size_t destPitch,
+  const void *src, size_t srcPitch,
+  size_t width, size_t height)
+----
+
+| Copies the data located at `src` to `dest`. The width of a row in bytes,
+including any padding, is specified by `destPitch` for `dest` and `srcPitch`
+for `src`. The amount of data to copy is specified as `width` bytes per row
+times `height` number of rows.  `width` must not be larger than either
+`srcPitch` or `destPitch`.
+
+|===
+
+=== Queue Shortcuts
+
+This extension also defines additional shortcut methods. The `queue` class
+gains three new methods:
+
+[cols="^60a,40"]
+|===
+| Member Function | Description
+
+|
+[source,c++]
+----
+event memcpy2D(void *dest, size_t destPitch,
+  const void *src, size_t srcPitch,
+  size_t width, size_t height)
+
+event memcpy2D(void *dest, size_t destPitch,
+  const void *src, size_t srcPitch,
+  size_t width, size_t height,
+  event depEvent)
+
+event memcpy2D(void *dest, size_t destPitch,
+  const void *src, size_t srcPitch,
+  size_t width, size_t height,
+  const std::vector<event> &depEvents)
+----
+
+| Copies the data located at `src` to `dest`.  The width of a row in bytes,
+including any padding, is specified by `destPitch` for `dest` and `srcPitch`
+for `src`.  The amount of data to copy is specified as `width` bytes per row
+times `height` number of rows.  `width` must not be larger than either `srcPitch`
+or `destPitch`.  If specified, the operation will not begin until the dependent
+events have completed.
+
+|===
+
+
+=== Feature test macro
```

Comment:
Kind of a nit, but the "Feature test macro" section comes first in the other extension specifications (i.e. first sub-section in  "Specification").  Other sub-sections follow that.

Reply:
Oops - this one was a little bit of a frankenstein bolting in the template rather than starting from it. Will update.